### ISSUE TITLE
R Session Aborted with render_highquality

### ISSUE THREAD

**Issue by wiesehahn** [tylermorganwall/rayshader#163]
when using render_highquality() I run into "R Session Aborted - R encountered a fatal error. The session was terminated."
e.g.:
```
#install.packages("rayshader")
#install.packages("rayrender")

library(rayshader)

montereybay %>%
  sphere_shade() %>%
  plot_3d(montereybay,zscale=50,zoom=0.6,theta=-90,phi=30)
render_highquality()
```
render_snapshot() runs without problems and also rayrender::render_scene seems to work.

R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

On another maschine there are no problems with render_highquality(), so which system settings could be responsible?

**Comment by tylermorganwall** [comment]
Works fine here, have you installed the latest versions from github?

```r
remotes::install_github("tylermorganwall/rayrender")
remotes::install_github("tylermorganwall/rayshader")
```

**Comment by wiesehahn** [comment]
yes, packages are up-to-date. 
I am trying to narrow it down. 

I tried to run the rayrender examples from the homepage and the first examples are all running smooth and CPU usage is high as expected (RAM usage is low though) e.g.:
```
library(rayrender)
scene = generate_ground(material=diffuse(checkercolor="grey20")) %>%
  add_object(sphere(y=0.2,material=glossy(color="#2b6eff",reflectance=0.05)))
render_scene(scene, parallel = TRUE, width = 800, height = 800, samples = 1000)
```

The first example which is not working is the following. In this case R is not crashing (at least I didnt wait that long), but it simply doesnt respond anymore. Monitoring CPU usage reveals that for some seconds CPU is used but falls back to 0% after some seconds for RStudio. 
```scene = generate_cornell()
render_scene(scene, lookfrom=c(278,278,-800),lookat = c(278,278,0), aperture=0, fov=40, samples = 1000,
             ambient_light=FALSE, parallel=TRUE, width=800, height=800, clamp_value = 5)
```
I tried to reduce the computational amount by reducing the samples and up until about 30 samples the result is finished quite quick, everthing above doesnt finish at all when `parallel = TRUE`. Setting `parallel = FALSE` solves this. 

And I am not sure if this is related to my original problem, since even reducing the number of samples dramatically results in "R Session Aborted" with render_highquality().

**Comment by wiesehahn** [comment]
I am still trying to figure out what causes the problem and I recognised that I run into this problem not only on a diffenerent maschine, but also on the same maschine outside my initial project. Hence, I guess it has to do with some dependency, although I am not yet sure which one.

I initialized a new project with renv, installed rayshader and rayrender and when trying to render highquality -> R Session aborted

Then I used renv::restore() with my lockfile from the other project and boom, no problem. Same on the other maschine. 

These are the packages that were updated:
```
The following package(s) will be updated:
# CRAN ===============================
- bslib        [0.2.5.1 -> 0.2.4]
- cachem       [1.0.5 -> 1.0.4]
- callr        [3.7.0 -> 3.6.0]
- cli          [2.5.0 -> 2.4.0]
- colorspace   [2.0-1 -> 2.0-0]
- ellipsis     [0.3.2 -> 0.3.1]
- hms          [1.1.0 -> 1.0.0]
- httpuv       [1.6.1 -> 1.5.5]
- jquerylib    [0.1.4 -> 0.1.3]
- knitr        [1.33 -> 1.32]
- later        [1.2.0 -> 1.1.0.1]
- pillar       [1.6.1 -> 1.6.0]
- processx     [3.5.2 -> 3.5.1]
- raster       [3.4-10 -> 3.4-5]
- rayshader    [0.24.10 -> 0.24.5]
- rgl          [0.106.8 -> 0.106.6]
- rlang        [0.4.11 -> 0.4.10]
- sass         [0.4.0 -> 0.3.1]
- stringi      [1.6.2 -> 1.5.3]
- tibble       [3.1.2 -> 3.1.0]
- vctrs        [0.3.8 -> 0.3.7]
- withr        [2.4.2 -> 2.4.1]
- xfun         [0.23 -> 0.22]
- BH           [* -> 1.75.0-0]
- DBI          [* -> 1.1.1]
- av           [* -> 0.6.0]
- classInt     [* -> 0.4-3]
- curl         [* -> 4.3]
- dplyr        [* -> 1.0.6]
- e1071        [* -> 1.7-6]
- generics     [* -> 0.1.0]
- ggplot2      [* -> 3.3.3]
- gridExtra    [* -> 2.3]
- gtable       [* -> 0.3.0]
- here         [* -> 1.0.1]
- isoband      [* -> 0.2.4]
- magick       [* -> 2.7.1]
- proxy        [* -> 0.4-25]
- rprojroot    [* -> 2.0.2]
- sf           [* -> 0.9-8]
- tidyselect   [* -> 1.1.1]
- units        [* -> 0.7-1]
- viridis      [* -> 0.6.0]
```

Some of them are not related to rayshader/rayrender but were simply used in other context in the other project. Before that I tried to update some packages individually (e.g.  cachem [1.0.5 -> 1.0.4],  callr [3.7.0 -> 3.6.0], rgl  [0.106.8 -> 0.106.6], ...) without success.

**Comment by hammao** [comment]
I'm facing exactly thesame problem.... i'm running on win10.

**Comment by akshaytanna** [comment]
@hammao 

I was facing the same issue. Fixed when I reinstalled both packages from github (Previously was using the CRAN package). I am running in Win10 as well.

**Comment by wiesehahn** [comment]
installing both packages from Github solved the problem.

**Comment by nahin29** [comment]
I updated rayshader from github. But still showing the error.

**Comment by yibril90** [comment]
I was using just fine the rayrender package till yesterday. By the time i ran this code section:

mat %>% 
  height_shade(texture = textura) %>%
  plot_3d (heightmap = matriz,
          **solid = FALSE**,
          soliddepth = 0,
          z = 1,
          shadowdepth = 0,
          windowsize = c(1000,1000), 
          phi = 90, 
          zoom = 1, 
          theta = 0, 
          background = "white")

Then I realized I wanted to see the 3d object with the surface rendered.

That simple change of the logical value from FALSE to TRUE was causing the issue you are mentioning. To solved I tried to install latest versions for both rayrender and rayshader packages from github. 

But it did not worked.

So I changed again to solid = FALSE and solved the chash of R Studio session.
